### PR TITLE
Positional magic

### PR DESCRIPTION
**Pull request by ebering**
This is a draft PR to track the implementation of [the positional magic experiment](https://github.com/crawl/crawl/wiki/Positional-Attack-Magic) and feedback thereon.

Current TODO (not in priority order):

- [x] Implement Eringya's Noxious Bog
- [x] Revise Corpse Rot
- [x] Take a hard look at Glaciate
- [x] Revise Spellforged Servitor
- [x] Starburst friendly fire checks
- [x] Dithmenos interactions
- [x] Refactor to use targeter class code for better friendly fire warnings &c.
- [x] Check out weird interactions with useless spell checks for meph/ignite poison
- [x] Add visual cloud trails to foxfire (EN0N)
- [x] Remove toxic bogs out of LOS
- [x] Save compatibility for players (spell library and memorized spells)
- [x] Save compatibility for monsters/ghosts

Needed tiles:

- [x] Eringya's Noxious Bog icon
- [x] Toxic Bog tile
- [x] Starburst icon
- [x] Hailstorm icon
- [x] Foxfire icon
- [x] Foxfire monster tile
- [x] IMB icon (Check that the old one isn't nonsense)
- [x] Dazzling Flash icon (Check the old one)

### CODE REVIEW

File: crawl-ref/source/dat/descript/spells.txt [hunk 1]
```diff
@@ -651,6 +651,15 @@ Greater Servant of Makhleb spell
 
 <Greater Servant of Makhleb ability>
 %%%%
+%%%%
+Hailstorm spell
+
+Conjures a cannonade of hail. The peculiarities of the magic invovled in
```
Comment (BacchusGDI) [inline crawl-ref/source/dat/descript/spells.txt]: Conjures a cannonade of hail. The peculiarities of the magic involved in

Comment (BacchusGDI) [inline crawl-ref/source/dat/descript/spells.txt]: Conjures a cannonade of hail. The peculiarities of the magic involved in

**Comment by ebering** [comment]
This is definitely an appreciated way of adding feedback (of the code cleanup/minor tweak kind). Thank you!

**Comment by kitchen-ace** [comment]
Corpse rot grammar error:

>and gathers the foul miasmic vapour around the caster. **The** eats away at the life

should probably be "It eats away at the life," or perhaps "The clouds eat away at the life."

**Pull request merged**